timbuctoo-instancev4/src/main/java/nl/knaw/huygens/timbuctoo/database/tinkerpop/Neo4jIndexHandler.java

```diff
@@ -66,31 +87,48 @@ public void addToOrUpdateQuickSearchIndex(Collection collection, String quickSea
   public void removeFromQuickSearchIndex(Collection collection, Vertex vertex) {
     Index<Node> index = getFulltextIndex(collection.getCollectionName());
 
-    index.remove(graphDatabase().getNodeById((long) vertex.id()), QUICK_SEARCH);
+    // make sure only this field is removed from the index.
+    index.remove(vertexToNode(vertex), QUICK_SEARCH);
   }
 
-  private GraphTraversal<Vertex, Vertex> traversalFromIndex(Collection collection, QuickSearch quickSearch) {
-    Index<Node> index = getFulltextIndex(getIndexName(collection));
-    IndexHits<Node> hits = index.query(QUICK_SEARCH, createQuery(quickSearch));
+  @Override
+  public GraphTraversal<Vertex, Vertex> findById(UUID timId) {
+    IndexHits<Node> hits = getExactIndex(ID_INDEX).query(TIM_ID, timId.toString());
     List<Long> ids = StreamSupport.stream(hits.spliterator(), false).map(h -> h.getId()).collect(toList());
```

Comment:
hmm, shouldn't this be at most 1 vertex? Since the quicksearch vs Id functions aren't the same signature anyway we might return just an Optional<vertex> instead of a GraphTraversal.